src/filter.jl

```diff
@@ -0,0 +1,62 @@
+"""
+    Filter([f])
+
+Uses a function `f` to identify values, rows, columns or slices of data that should be
+removed during an `apply` call. The default function `f` will removing `missing`s, or any
+rows, columns or slices containing `missing`s.
+"""
+struct Filter
+    func::Function
+end
+
+Filter() = Filter(_keep)
+
+_keep(x) = !ismissing(x)
+_keep(x::Union{Tuple, AbstractArray, IterTools.PropertyValues}) = all(_keep, x)
+
+apply!(data::Vector, f::Filter) = Base.filter!(f.func, data)
+function apply!(data::Vector{<:NamedTuple}, f::Filter; dims=:rows)
+    d = dim(data, dims)
+    d == 1 || throw(ArgumentError("Rowtables only support in-place filtering rowwise."))
+    return Base.filter!(r -> f.func(propertyvalues(r)), data)
+end
+
+apply(data::Vector, f::Filter) = Base.filter(f.func, data)
+function apply(data::Vector{<:NamedTuple}, f::Filter; dims=:rows)
+    d = dim(data, dims)
+    if d == 1
+        return Base.filter(r -> f.func(propertyvalues(r)), data)
+    else
+        return materializer(data)(apply(Tables.columns(data), f; dims=dims))
+    end
+end
+
+function apply(data::AbstractArray{Union{T, Missing}}, f::Filter; dims) where T
+    d = dim(data, dims)
+    mask = map(f.func, eachslice(data; dims=d))
+    idx = (i == d ? mask : Colon() for i in 1:ndims(data))
+    return data[idx...]
```

Comment:
This feels like `base.selectdims` but I am not sure if it actual is or not

Reply:
Yep, looks like `selectdim` is what I want 👍